FordLabs/retroquest
# Publish metrics to StatsD

## Overview
Publishes RetroQuest's current metrics to StatsD.

Currently, RetroQuest's metrics can only be obtained by making an admin-authenticated request to the metrics endpoint.

This PR will add custom Micrometer metrics tracking for these metrics so that RetroQuest will publish them to StatsD. 

### Notes
I made some fairly substantial refactors to the Metrics service to push calculations to the database in order to facilitate more frequent polling. 

## Testing Instructions
The existing tests should continue to pass, and no behavioral changes are expected as a result of this change. Assertions on metrics values were added to existing API tests that would cause a change in those values.
